scalalib/src/mill/scalalib/JlinkModule.scala

```diff
@@ -0,0 +1,90 @@
+package mill
+package scalalib
+
+import mill._
+
+/**
+ * Support building modular runtime images with the `jlink` tool, which is included in JDK 9 and later.
+ * 
+ * The official `jlink` docs: https://docs.oracle.com/en/java/javase/23/docs/specs/man/jlink.html
+ */
+trait JlinkModule extends JavaModule {
+
+  /** The base name for the runtime image */
+  def jlinkImageName: T[String] = T { "jlink" }
+
+  /** Name of the main module to be included in the runtime image */
+  def jlinkModuleName: T[String] = T { "" }
+
+  /** The main module's version number. */
+  def jlinkModuleVersion: T[Option[String]] = T { None }
+
+  /** The main class to use as the runtime entry point. */
+  def jlinkMainClass: T[String] = T { finalMainClass() }
+
+  /** Compress level for the runtime image. 
+   * Valid values range between: 
+   *  "zip-0" (no compression) and "zip-9" (best compression)
+   * Defaults to "zip-6" */
+  def jlinkCompressLevel: T[String] = T { "zip-6" }
```

Comment:
This seems to be having problems in CI; are there any other values we could use?

Reply:
On my machine, I have OpenJDK installed:
```sh
java --version
openjdk 21.0.1 2023-10-17
OpenJDK Runtime Environment (build 21.0.1+12-29)
OpenJDK 64-Bit Server VM (build 21.0.1+12-29, mixed mode, sharing)
```
But a momentary switch to Oracle's JDK shows that `jlink` will only accept either of [`0`, `2`] as valid values for compression.

To summarize:
* On Oracle's build of the JDK, valid values are: `0` (no compression), `1` and `2` (zip compression).
* On recent builds of OpenJDK, valid values range between: [`"zip-0"` - `"zip-9"`].

I've updated the Usage examples so this subtle difference between JDK distributions is clearer.